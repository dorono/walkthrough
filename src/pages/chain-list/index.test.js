import React from 'react';
import {shallow} from 'enzyme';
import {ChainListPage} from './index';

jest.mock('../../utils/date', () => {
    return {
        currentTimezone: () => '11/11/11',
        formatDateLong: () => '11111111',
    };
});

const defaultProps = {
    data: [{chain_id: 1}],
    location: {},
    count: 0,
    limit: 0,
    offset: 0,
};

describe('ChainList', () => {
    it('should render without errors', () => {
        shallow(<ChainListPage {...defaultProps} />);
    });

    it('should match snapshot', () => {
        expect(shallow(<ChainListPage {...defaultProps} />)).toMatchSnapshot();
    });
});
