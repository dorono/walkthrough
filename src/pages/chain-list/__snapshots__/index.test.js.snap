// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChainList should match snapshot 1`] = `
<Container
  primary={true}
  showFullWidthBanner={false}
  title="Chains"
>
  <withRouter(Table)
    columns={
      Array [
        "CREATED (UTC-0500)",
        "CHAIN ID",
        "EXTERNAL IDS",
      ]
    }
    ellipsis={
      Array [
        1,
        2,
      ]
    }
    responsive={true}
    rows={
      Array [
        Object {
          "chain_id": 1,
        },
      ]
    }
  >
    <Component />
  </withRouter(Table)>
  <PendingLegend
    fullWidthBannerText="This chain has not yet been written to the Factom Blockchain."
    highlightBanner={false}
    show={false}
  />
  <Pagination
    count={0}
    limit={0}
    offset={0}
  />
</Container>
`;
